modules/storage/cinder.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2022 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// CinderExtraVolTypeUndefined can be used to label an extraMount which
+	// is not associated with a specific backend
+	CinderExtraVolTypeUndefined ExtraVolType = "Undefined"
+	// CinderExtraVolTypeCeph can be used to label an extraMount which
+	// is associated to a Ceph backend
+	CinderExtraVolTypeCeph      ExtraVolType = "Ceph"
+
+	// CinderBackup is the definition of the cinder-backup group
+	CinderBackup    ServiceType = "CinderBackup"
+	// CinderVolume is the definition of the cinder-volume group
+	CinderVolume    ServiceType = "CinderVolume"
+	// CinderScheduler is the definition of the cinder-scheduler group
+	CinderScheduler ServiceType = "CinderScheduler"
+	// CinderAPI is the definition of the cinder-api group
+	CinderAPI       ServiceType = "CinderAPI"
+
+	// Cinder is the global ServiceType that refers to all the components deployed
+	// by the cinder operator
+	Cinder ServiceType = "Cinder"
+)
+
+// CinderExtraVolMounts exposes additional parameters processed by the cinder-operator
+// and defines the common VolMounts structure provided by the main storage module
+type CinderExtraVolMounts struct {
+	// +kubebuilder:validation:Optional
+	Name string `json:"name,omitempty"`
+	// +kubebuilder:validation:Optional
+	Region string `json:"region,omitempty"`
+	// +kubebuilder:validation:Required
+	VolMounts []VolMounts `json:"extraVol"`
+}
+
+// AppendVolumeMount is a function that appends a valid Volume to the existing corev1.VolumeMount array
+func (c *CinderExtraVolMounts) AppendVolumeMount(vm []corev1.VolumeMount, extraVol []CinderExtraVolMounts, svc []ServiceType) []corev1.VolumeMount {
+
+	if len(extraVol) > 0 {
```

Comment:
This feels unnecessary. If `extraVol` is an empty list then the below for loop will be noop anyway.